src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/isl/IslFsm.java

```diff
@@ -343,9 +345,13 @@ private void sendInstallMultiTable(IIslCarrier carrier, Endpoint ingress, Endpoi
         }
     }
 
-    private void sendEnableExhaustedPollMode(IIslCarrier carrier) {
-        carrier.exhaustedPollModeUpdateRequest(reference.getSource(), true);
-        carrier.exhaustedPollModeUpdateRequest(reference.getDest(), true);
+    private void sendEnableExhaustedPollMode(IIslCarrier carrier, Endpoint endpoint) {
+        Collection<Isl> isls = islRepository.findByEndpoint(endpoint.getDatapath(), endpoint.getPortNumber()).stream()
```

Comment:
This is definitely an incorrect solution. One ISL object/handler must not interfere with another. Plus race condition (it almost impossible in an environment where delete is done manually, but if it will be done by another application chances of the race quickly become realistic).

Perhaps we should make step backward and do not control exhausted mode from IslFsm, but return this control into UniIslHandler. I propose to "reset" status of UniIslHandler at the end of ISL removal process. So, after reset, when UniIslHandler receive first FAIL response on discovery request it will turn on exausted mode by itself.

Some point to uni-isl reset processing:
* reset request mut include IslReference - so UniIslHandler can compare it with it's own and perform reset only if reference match (do not reset already "moved" handler).
* discovery poll mode must be switched to "generic" mode.

Reply:
A very good proposal. I've made changes according to it.